genesyscloud/integration_facebook/genesyscloud_integration_facebook_proxy.go

```diff
@@ -0,0 +1,168 @@
+	return p.updateIntegrationFacebookAttr(ctx, p, id, integrationFacebook)
+}
+
+// deleteIntegrationFacebook deletes a Genesys Cloud integration facebook by Id
+func (p *integrationFacebookProxy) deleteIntegrationFacebook(ctx context.Context, id string) (response *platformclientv2.APIResponse, err error) {
+	return p.deleteIntegrationFacebookAttr(ctx, p, id)
+}
+
+// createIntegrationFacebookFn is an implementation function for creating a Genesys Cloud integration facebook
+func createIntegrationFacebookFn(ctx context.Context, p *integrationFacebookProxy, integrationFacebook *platformclientv2.Facebookintegrationrequest) (*platformclientv2.Facebookintegration, *platformclientv2.APIResponse, error) {
+	return p.conversationsApi.PostConversationsMessagingIntegrationsFacebook(*integrationFacebook)
+}
+
+// getAllIntegrationFacebookFn is the implementation for retrieving all integration facebook in Genesys Cloud
+func getAllIntegrationFacebookFn(ctx context.Context, p *integrationFacebookProxy) (*[]platformclientv2.Facebookintegration, *platformclientv2.APIResponse, error) {
+	var allFacebookIntegrationRequests []platformclientv2.Facebookintegration
+
+	facebookIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsFacebook(1, 100, "", "", "")
+	if err != nil {
+		return nil, resp, fmt.Errorf("Failed to get facebook integration request: %v %v", err, resp)
+	}
+	if facebookIntegrationRequests.Entities == nil || len(*facebookIntegrationRequests.Entities) == 0 {
+		return &allFacebookIntegrationRequests, resp, nil
+	}
+
+	allFacebookIntegrationRequests = append(allFacebookIntegrationRequests, *facebookIntegrationRequests.Entities...)
+
+	for pageNum := 2; pageNum <= *facebookIntegrationRequests.PageCount; pageNum++ {
+		const pageSize = 100
+
+		facebookIntegrationRequests, resp, err := p.conversationsApi.GetConversationsMessagingIntegrationsFacebook(pageNum, pageSize, "", "", "")
+		if err != nil {
+			return nil, resp, fmt.Errorf("Failed to get facebook integration request: %v %v", err, resp)
+		}
+
+		if facebookIntegrationRequests.Entities == nil || len(*facebookIntegrationRequests.Entities) == 0 {
+			break
+		}
+
+		allFacebookIntegrationRequests = append(allFacebookIntegrationRequests, *facebookIntegrationRequests.Entities...)
+	}
+
+	return &allFacebookIntegrationRequests, resp, err
+}
+
+// getIntegrationFacebookIdByNameFn is an implementation of the function to get a Genesys Cloud integration facebook by name
+func getIntegrationFacebookIdByNameFn(ctx context.Context, p *integrationFacebookProxy, name string) (string, bool, *platformclientv2.APIResponse, error) {
+	facebookIntegrationRequests, resp, err := getAllIntegrationFacebookFn(ctx, p)
+	if err != nil {
+		return "", false, resp, err
+	}
+
+	if facebookIntegrationRequests == nil || len(*facebookIntegrationRequests) == 0 {
+		return "", true, resp, fmt.Errorf("No integration facebook found with name %s", name)
```

Comment:
can distinguish this error with that on line 152. This error can suggest there are no facebookIntegrationRequests at all.